File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.py
class Solution(object):
    def searchRange(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        
        l=0
        r=len(nums)-1
        mid=0
        ans1=ans2=-1
        while l <= r:
            mid = (l + r) // 2
            if nums[mid] < target:
                l = mid + 1
            elif nums[mid] >= target:
                r = mid - 1
                if nums[mid] == target:
                    ans2 = mid

        l = 0
        r = len(nums) - 1
        while l <= r:
            mid = (l + r) // 2
            if nums[mid] <= target:
                l = mid + 1
                if nums[mid] == target:
                    ans1 = mid
            else:
                r = mid - 1
        
        if  ans1!=-1 and ans2!=-1:
            return [ans2,ans1]
        else:
            return [-1,-1]
        

File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.java
import java.util.ArrayList;
import java.util.List;

class Solution {
    public int[] searchRange(int[] nums, int target) {
        int[] result = new int[]{-1, -1}; // Initialize the result with default values
        
        // Find lower bound
        int l = 0, r = nums.length - 1;
        int lowerBound = -1;
        while (l <= r) {
            int mid = l + (r - l) / 2;
            if (nums[mid] >= target) {
                if (nums[mid] == target) {
                    lowerBound = mid;
                }
                r = mid - 1;
            } else {
                l = mid + 1;
            }
        }
        
        // If the lower bound is not found, return [-1, -1]
        if (lowerBound == -1) {
            return result;
        }
        
        // Find upper bound
        l = 0;
        r = nums.length - 1;
        int upperBound = -1;
        while (l <= r) {
            int mid = l + (r - l) / 2;
            if (nums[mid] <= target) {
                if (nums[mid] == target) {
                    upperBound = mid;
                }
                l = mid + 1;
            } else {
                r = mid - 1;
            }
        }
        
        result[0] = lowerBound;
        result[1] = upperBound;
        
        return result;
    }
}
